spec.html

```diff
@@ -48962,88 +48934,89 @@ <h1>agent-order</h1>
+        <li>For each pair (_R_, _W_) in SharedDataBlockEventSet(_execution_), _R_ reads-from _W_ in _execution_ if reads-bytes-from(_R_) in _execution_ contains _W_.</li>
       </ul>
     </emu-clause>
 
-    <emu-clause id="sec-host-synchronizes-with" aoid="host-synchronizes-with">
+    <emu-clause id="sec-host-synchronizes-with">
       <h1>host-synchronizes-with</h1>
-      <p>For a candidate execution _execution_, _execution_.[[HostSynchronizesWith]] is a host-provided strict partial order on host-specific events that satisfies at least the following.</p>
+      <p>For a candidate execution _execution_, its <dfn>host-synchronizes-with</dfn> Relation is a host-provided strict partial order on host-specific events that satisfies at least the following.</p>
       <ul>
-        <li>If _execution_.[[HostSynchronizesWith]] contains (_E_, _D_), _E_ and _D_ are in HostEventSet(_execution_).</li>
-        <li>There is no cycle in the union of _execution_.[[HostSynchronizesWith]] and _execution_.[[AgentOrder]].</li>
+        <li>If _E_ host-synchronizes-with _D_ in _execution_, _E_ and _D_ are in HostEventSet(_execution_).</li>
+        <li>There is no cycle in the union of host-synchronizes-with and is-agent-order-before in _execution_.</li>
       </ul>
 
       <emu-note>
-        <p>For two host-specific events _E_ and _D_, _E_ host-synchronizes-with _D_ implies _E_ happens-before _D_.</p>
+        <p>For two host-specific events _E_ and _D_ in a candidate execution _execution_, _E_ host-synchronizes-with _D_ in _execution_ implies _E_ happens-before _D_ in _execution_.</p>
       </emu-note>
       <emu-note>
-        <p>The host-synchronizes-with relation allows the host to provide additional synchronization mechanisms, such as `postMessage` between HTML workers.</p>
+        <p>This Relation allows the host to provide additional synchronization mechanisms, such as `postMessage` between HTML workers.</p>
       </emu-note>
     </emu-clause>
 
-    <emu-clause id="sec-synchronizes-with" aoid="synchronizes-with">
+    <emu-clause id="sec-synchronizes-with">
       <h1>synchronizes-with</h1>
-      <p>For a candidate execution _execution_, _execution_.[[SynchronizesWith]] is the least Relation on events that satisfies the following.</p>
+      <p>For a candidate execution _execution_, its <dfn>synchronizes-with</dfn> Relation is the least Relation on events that satisfies the following.</p>
       <ul>
         <li>
-          For each pair (_R_, _W_) in _execution_.[[ReadsFrom]], _execution_.[[SynchronizesWith]] contains (_W_, _R_) if _R_.[[Order]] is ~seq-cst~, _W_.[[Order]] is ~seq-cst~, and _R_ and _W_ have equal ranges.
+          For each pair (_R_, _W_) in reads-from in _execution_, _W_ synchronizes-with _R_ in _execution_ if _R_.[[Order]] is ~seq-cst~, _W_.[[Order]] is ~seq-cst~, and _R_ and _W_ have equal ranges.
```

Comment:
This would read better to me, and is more directly defining `synchronizes-with`:

```suggestion
          For each event _R_ and _W_, _W_ synchronizes-with _R_ in _execution_ if and only if _R_ reads-from _W_ in _execution_,  _R_.[[Order]] is ~seq-cst~, _W_.[[Order]] is ~seq-cst~, and _R_ and _W_ have equal ranges.
```

Reply:
Agreed. I've opted for "For events _R_ and _W_" instead of "For each event _R_ and _W_".